components/basketball/matches/MatchList.tsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { formatDateTime } from "@/utils/formatDateTime";
+
+// Define props interface
+interface MatchListProps {
+  matches: BbSeasonMatch[];
+  timeOption: string;
+  teamOption: string;
+}
+
+const MatchList: React.FC<MatchListProps> = ({ matches, timeOption, teamOption }) => {
+
+    const filterMatches = (matches: any[], timeOption: string, teamOption: string) => {
```

Comment:
instead of doing a client side filtering, I would recommend to explore a API solution to `refetch` the data once filter changed